nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/test/java/org/apache/nifi/oauth2/JWTBearerOAuth2AccessTokenProviderTest.java

```diff
@@ -0,0 +1,300 @@
+import com.nimbusds.jwt.JWTClaimsSet;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.PropertyValue;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.key.service.api.PrivateKeyService;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.web.client.api.HttpResponseEntity;
+import org.apache.nifi.web.client.api.WebClientService;
+import org.apache.nifi.web.client.provider.api.WebClientServiceProvider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.net.URISyntaxException;
+import java.security.PrivateKey;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class JWTBearerOAuth2AccessTokenProviderTest {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private ConfigurationContext mockContext;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private ValidationContext mockValidationContext;
+
+    @Mock
+    private PrivateKeyService mockKeyService;
+
+    @Mock
+    private WebClientServiceProvider mockWebClientServiceProvider;
+
+    @Mock
+    private WebClientService mockWebClientService;
+
+    @Mock
+    private HttpResponseEntity mockResponseEntity;
+
+    @Mock
+    private ComponentLog mockLogger;
+
+    private JWTBearerOAuth2AccessTokenProviderForTests provider;
+
+    @BeforeEach
+    void setUp() {
+        // Initialize mocks
+        MockitoAnnotations.openMocks(this);
+        provider = new JWTBearerOAuth2AccessTokenProviderForTests();
+
+        // Mock PropertyValue for WEB_CLIENT_SERVICE
+        PropertyValue webClientServicePropertyValue = mock(PropertyValue.class);
+        when(mockContext.getProperty(JWTBearerOAuth2AccessTokenProvider.WEB_CLIENT_SERVICE))
+                .thenReturn(webClientServicePropertyValue);
+
+        // Mock WebClientService
+        when(webClientServicePropertyValue.asControllerService(WebClientServiceProvider.class))
+                .thenReturn(mockWebClientServiceProvider);
+
+        when(mockWebClientServiceProvider.getWebClientService()).thenReturn(mockWebClientService);
+    }
+
+    @Test
+    void testClaimsAreSetProperly() throws Exception {
+        // Setup context
+        setContext();
+
+        setPrivateKeyMock(RSAPrivateKey.class);
+        when(mockContext.getProperty(JWTBearerOAuth2AccessTokenProvider.SIGNING_ALGORITHM).getValue()).thenReturn(JWSAlgorithm.RS256.getName());
```

Comment:
Instead of mocking the context calls, it seems like it would be cleaner to use the TestRunner.

Reply:
Pushed a commit where I'm using the test runner instead. Not ideal as I can't call the customValidate directly and specifically check what is making the controller service invalid but that could be a follow up improvement in the test framework.